components/SplitPane/demo/src/jvmMain/kotlin/org/jetbrains/compose/splitpane/demo/Main.kt

```diff
@@ -24,40 +41,69 @@ fun main() = singleWindowApplication(
     MaterialTheme {
         val splitterState = rememberSplitPaneState()
         val hSplitterState = rememberSplitPaneState()
-        HorizontalSplitPane(
-            splitPaneState = splitterState
-        ) {
-            first(20.dp) {
-                Box(Modifier.background(Color.Red).fillMaxSize())
-            }
-            second(50.dp) {
-                VerticalSplitPane(splitPaneState = hSplitterState) {
-                    first(50.dp) {
-                        Box(Modifier.background(Color.Blue).fillMaxSize())
-                    }
-                    second(20.dp) {
-                        Box(Modifier.background(Color.Green).fillMaxSize())
-                    }
+        var delta by remember { mutableStateOf("20") }
+        var percentage by remember { mutableStateOf("0.20") }
+        Row {
+            Column(verticalArrangement = Arrangement.spacedBy(10.dp), modifier = Modifier.padding(10.dp).width(180.dp)) {
+                Text("Action panel", fontWeight = FontWeight.Bold)
+                Button(onClick = { splitterState.moveEnabled = !splitterState.moveEnabled }) {
+                    Text(if (splitterState.moveEnabled) "Freeze V" else "Unfreeze V")
+                }
+                Button(onClick = { hSplitterState.moveEnabled = !hSplitterState.moveEnabled }) {
+                    Text(if (hSplitterState.moveEnabled) "Freeze H" else "Unfreeze H")
+                }
```

Comment:
These could be checkboxes instead.

Reply:
Looks fine either way

Reply:
I've just find time to finally improve it. But yeah, looks fine either way for me as well.